ng/manage.ts

```diff
@@ -576,6 +642,52 @@ angular.module("armExplorer", ["ngRoute", "ngAnimate", "ngSanitize", "ui.bootstr
 
     initUser();
 
+    initResourceSearch();
+
+    function initResourceSearch(): void {
+        $scope.resourceSearchModel = <IResourceSearch>{
+            isSuggestListDisplay: false,
+            suggestions: []
+        };
+
+        refreshResourceSearchCache();
+
+        // hide suggestion list when user click somewhere else
+        $("body").click(function (event) {
+            if (event && event.target
+                && event.target.getAttribute("id") !== "resource-search-input"
+                && !$.contains($("#resource-search-input")[0], event.target)
+                && event.target.getAttribute("id") !== "resource-search-list"
+                && !$.contains($("#resource-search-list")[0], event.target)) {
+
+                $scope.resourceSearchModel.isSuggestListDisplay = false;
+            }
+        });
+    }
+
+    // ARM not accept filter, query all data and perform client filter for now
+    var isResourceSearchCacheRefreshing: boolean = false;
+    function refreshResourceSearchCache(): void {
+        // only refresh cache if cache wasn`t there or too old
+        if (!isResourceSearchCacheRefreshing
+            && (!$scope.resourceSearchCache || Date.now() - $scope.resourceSearchCache.timestamp > 10000)) {
+            console.log("refresh cache ........");
```

Comment:
removed. this log was debugging, to tell me only one request will be fire at a time